cluster_migration_core/cluster_migration_core/cluster_management/docker_framework_client.py

```diff
@@ -113,16 +141,35 @@ def remove_volume(self, volume: Volume):
         self.logger.debug(f"Removed volume {volume.name}")
 
     def create_container(self, image: str, container_name: str, network: Network, ports: List[PortMapping],
-                         volumes: List[DockerVolume], ulimits: List[Ulimit], env_variables: Dict[str, str]
+                         volumes: List[DockerVolume], ulimits: List[Ulimit], env_variables: Dict[str, str] = None,
+                         extra_hosts: Dict[str, str] = None, detach: bool = True
                          ) -> Container:
 
         # TODO - need handle if container already exists (name collision)
         # TODO - need handle if we exceed the resource allocation for Docker
         # TODO - need handle port contention
 
+        # Initialize optional, mutable containers
+        if not env_variables:
+            env_variables = {}
+        if not extra_hosts:
+            extra_hosts = []
```

Comment:
what is this for?

Reply:
I recently learned that initializing default value for mutable containers in the argument list of a methods is dangerous.

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/